lib/client/keyagent.go

```diff
@@ -240,12 +267,19 @@ func (a *LocalKeyAgent) CheckHostSignature(hostId string, remote net.Addr, key s
 	log.Debugf("[KEY AGENT] got %d known hosts", len(keys))
 	for i := range keys {
 		if sshutils.KeysEqual(cert.SignatureKey, keys[i]) {
-			log.Debugf("[KEY AGENT] verified host %s", hostId)
+			log.Debugf("[KEY AGENT] verified host %s", host)
 			return nil
 		}
 	}
-	err = trace.AccessDenied("untrusted host %v", hostId)
-	log.Error(err)
+	// final step: lets ask user:
```

Comment:
Can you update the comment here to something like: "final step: we have not seen this host certificate before, prompt the user, then add to known hosts".

On my first pass I was a little confused why we were always asking the user in the final step when they had a certificate. I missed the `return nil` a few lines up because typically you return an error in the body and `return nil` (success) at the end.